libs/filameshio/src/MeshReader.cpp

```diff
@@ -159,30 +159,32 @@ static size_t fileSize(int fd) {
 
 namespace filamesh {
 
-MeshReader::Mesh MeshReader::loadMeshFromFile(filament::Engine* engine, const utils::Path& path,
-        MaterialRegistry& materials) {
-
+MeshReader::Mesh MeshReader::loadMeshFromFile(filament::Engine *const engine, const utils::Path& path,
+        MaterialRegistry& materialRegistry) {
     Mesh mesh;
 
-    int fd = open(path.c_str(), O_RDONLY);
+    /* Open the mesh file for reading only. */
+    const int fd = open(path.c_str(), O_RDONLY);
+
+    /* Check if an error occurred when opening the file. */
+    if (fd == -1) return mesh;
 
-    size_t size = fileSize(fd);
-    char* data = (char*) malloc(size);
-    read(fd, data, size);
+    const size_t size = fileSize(fd);
+    const char *const data = (char *)malloc(size);
 
     if (data) {
-        char *p = data;
-        char *magic = p;
-        p += sizeof(MAGICID);
+        (void)read(fd, (void *)data, size);
```

Comment:
That's not a style we're using anywhere. Why not use the return value and do the `strncmp` only if successful? Or, at least `assert()` if not.

Reply:
OK.